### ISSUE TITLE
Declaring context type on class component breaks static-property-placement

### ISSUE THREAD

**Issue by klimashkin** [yannickcr/eslint-plugin-react#3467]
Hi,

After upgrading `eslint-plugin-react` from 7.31.8 to 7.31.9+ (#2581), it started breaking the `declare` keyword on context in TypeScript:

<img width="800" alt="image" src="https://user-images.githubusercontent.com/576077/196576130-078e1d84-434e-46cc-a76c-79d2c28dea18.png">

Not sure what is the mitigation in this case

**Comment by ljharb** [comment]
The rule has this explicit code:
```js
    // Flow support
    if (node.typeAnnotation && node.key.name === 'context') {
      return true;
    }
```

We don't seem to have tests for it, though, and I don't see this actual direct code in 54b738d24 / #1526 - @jomasti, any idea?

**Comment by OlivierKamers** [comment]
I ran into the same problem and started digging into the source code of lint rules for the first time. I don't understand enough yet to propose a fix but if it helps this could be a test case to reproduce it, which fails if I add it to the `valid` test cases of this rule:

```javascript
{
  code: `
    class MyComponent extends React.Component {
      static contextTypes = {
        something: PropTypes.bool
      };
      declare context: MyContext;
    }
  `,
  features: ['ts'],
  options: [STATIC_PUBLIC_FIELD],
},
```

I will look around a bit more. Even if I don't find a fix it's a good learning experience!

**Comment by OlivierKamers** [comment]
@ljharb the snippet you mentioned is indeed identifying the `context` declaration as a `contextTypes` declaration. 
From what I find it doesn't look like it's something that should be supported in Flow either (https://github.com/facebook/flow/issues/7166, https://github.com/facebook/flow/issues/2916) as it's used in conjunction with `contextTypes`. 

Removing this check fixes the failing tests I added and doesn't break any other tests, but as you said they were just missing so there's a risk that it breaks something else. 

Another possibility is checking if there is already a sibling that looks like a `contextTypes` declaration, but it feels weird and not robust.
```javascript
    // Flow support
    if (node.typeAnnotation && node.key.name === 'context') {
      return !node.parent.body.some((sibling) => isContextTypesDeclaration(sibling));
    }
```

**Comment by ljharb** [comment]
Yeah, that doesn't feel like a safe approach. You might be right that it should never have been added, and that it's safe to remove, but I'm hesitant to do that without hearing from @jomasti.

**Comment by ljharb** [comment]
Sounds good, let’s remove it
